### PR TITLE
Improve our configuration of `ClassNaming` and `FunctionNaming`

### PR DESCRIPTION
**Pull request by BraisGabin**
Related with #5903

### CODE REVIEW

File: config/detekt/detekt.yml [hunk 1]
```diff
@@ -90,17 +90,11 @@ formatting:
...
-    excludes:
-      - '**/test/**'
-      - '**/androidTest/**'
-      - '**/commonTest/**'
-      - '**/functionalTest/**'
-      - '**/jvmTest/**'
-      - '**/jsTest/**'
-      - '**/iosTest/**'
+    excludes: []
+    ignoreAnnotated: ['Test', 'ParameterizedTest', 'RepeatedTest', 'TestFactory']
```
Comment (TWiStErRob) [inline config/detekt/detekt.yml:97]: Let's go with FQCNs like:

https://github.com/TWiStErRob/net.twisterrob.cinema/blob/69111724bf4f828b3b1d9980222d643e58c5aa3a/Heroku/config/detekt/detekt.yml#L90-L101

File: config/detekt/detekt.yml [hunk 2]
```diff
@@ -90,17 +90,11 @@ formatting:
 
 naming:
   ClassNaming:
-    excludes: ['**/*Spec.kt']
+    ignoreAnnotated: ['Nested']
   FunctionNaming:
```
Comment (TWiStErRob) [inline config/detekt/detekt.yml:94]: speaking of https://github.com/TWiStErRob/net.twisterrob.cinema/blob/69111724bf4f828b3b1d9980222d643e58c5aa3a/Heroku/config/detekt/detekt.yml#L90-L101, do you want to update that rule too?

Comment (BraisGabin) [reply config/detekt/detekt.yml:94]: We have `FunctionMaxLength` disabled AFAIK so there is not need to change our configuration for it.

File: config/detekt/detekt.yml (hunk 1 above)
Comment (BraisGabin) [reply config/detekt/detekt.yml:97]: In this case I think that I prefer no to use FQCNs. They are easier to read. I don't see how an annotation called `ParametrizedTest` is not related with a test function.

Comment (TWiStErRob) [reply config/detekt/detekt.yml:97]: What about nested ([Gradle](https://docs.gradle.org/current/javadoc/org/gradle/api/tasks/Nested.html)) and testfactory? Those could be anything, and only the JUnit 5 variants should be allowed more relaxed names.

It's shorter to read and write, yes, but if you actually want to understand why and what it is, then the implicitness requires way more brainpower. :)

Comment (BraisGabin) [reply config/detekt/detekt.yml:97]: Agree `Nested` is way too generic. Changed.

**Review by cortinico — APPROVED** [review]

**Pull request merged**